src/vm/i386/cgenx86.cpp

```diff
@@ -299,7 +299,12 @@ void TransitionFrame::UpdateRegDisplay(const PREGDISPLAY pRD)
 
     MethodDesc * pFunc = GetFunction();
     _ASSERTE(pFunc != NULL);
+
+#ifdef WIN64EXCEPTIONS
+    UpdateRegDisplayHelper(pRD, 0);
```

Comment:
@janvorli You're right. It would be better to use FEATURE_PAL (instead of WIN64EXCEPTIONS) here.

As discussed in #9272, libunwind does not consider calling convention at all, and stack frames from native method such as ``ThePreStub`` is unwound by libunwind.

As a result, stack walker failed to find the target frame as the esp comes from libuwnind and the esp updated by this method are different.